Jenkinsfile.promote

```diff
@@ -275,6 +305,29 @@ pipeline {
     }
 }
 
+void prNextSnapshotVersion(String repo, String sourceBranch, String targetBranch){
+    String nextMicroSnapshotVersion = getNextMicroSnapshotVersion(getProjectVersion())
+    dir("$repo-bot") {
+        prepareForPR(repo, sourceBranch)
+
+        maven.mvnSetVersionProperty("version.org.optaplanner", nextMicroSnapshotVersion)
+        maven.mvnVersionsUpdateParentAndChildModules(nextMicroSnapshotVersion, true)
+        gradleVersionsUpdate(repo, nextMicroSnapshotVersion)
+
+        String prLink = commitAndCreatePR("[${sourceBranch}] Update snapshot version to ${nextMicroSnapshotVersion}",
+                {
+                    githubscm.findAndStageNotIgnoredFiles('pom.xml')
+                    githubscm.findAndStageNotIgnoredFiles('build.gradle')
+                }, sourceBranch, targetBranch)
+
+        setPipelinePrLink(repo, prLink)
```

Comment:
calling the method twice with same repo will do an override. no impact here as everything happens in same method but could be in the future if we want to reuse the links somewhere else ...